app/services/current-device.js

```diff
@@ -22,4 +22,43 @@ export default class CurrentDeviceService extends Service {
 
     return device;
   }
+
+  async configure(ssid, password) {
+    const response = await fetch(
+      `${ENV.APP.BASE_API_URL}/api/v1/device/${this.data.serialNumber}/configure`,
+      {
+        method: "POST",
+        body: JSON.stringify({
+          configuration: {
+            ssid,
+            password,
+          },
+        }),
+      }
+    );
+
+    if (!response.ok) {
+      const text = await response.text();
+      throw new Error(text);
+    }
+
+    const data = await response.json();
+    const {
+      configuration: {
+        encryption,
+        hidden,
+        ssid: setSsid,
+        password: setPassword,
+      },
+    } = data;
+    this.data = {
+      ...this.data,
+      ssid: setSsid,
+      encryption,
+      password: setPassword,
+      isHidden: hidden,
+    };
```

Comment:
I find this bit hard to read, what do you think about this instead?
```suggestion
    this.data = {
      ...this.data,
      ssid: data.configuration.ssid,
      encryption: data.configuration.encryption,
      password: data.configuration.password,
      isHidden: data.configuration.hidden,
    };
```